### PR TITLE
Fix To Manually Providing Toggle State and Function For Small Navigation Bar

### PR DESCRIPTION
**Pull request by ErnestDaDev**
The reason I did not want the value and function to be passed together like a useState function is to allow room for control over the toggle function.

### CODE REVIEW

File: src/components/ResponsiveNavbar/ResponsiveNavbar.tsx [hunk 1]
```diff
@@ -98,10 +99,26 @@ export const ResponsiveNavbar = (props: ResponsiveNavbarProps) => {
         )}
         {children ?? (
           <ul>
-            <li>ABOUT</li>
-            <li>PROJECTS</li>
-            <li>ELEMENTS</li>
-            <li>CONTACT</li>
+            <li  onClick={() => {
```
Comment (Abusayid693) [inline src/components/ResponsiveNavbar/ResponsiveNavbar.tsx:102]: This approach can be problematic, coz route change might take some time, So immediately closing navigation (using onClick handler) even before routing completed will give bad user experience.

**Better solution would be to use routing events:**
```ts
const router = useRouter();
useEffect(() => {
  router.events.on('routeChangeComplete', setClose)
  return () => {
    router.events.off('routeChangeComplete', setClose)
  };
}, [router.events]);
```

But that would be framework specific, we don't need to take care, manual control is enough for now.

**Review by Abusayid693 — APPROVED** [review]

**Comment by Abusayid693** [comment]
@ErnestDaDev Thanks for the PR

File: src/components/ResponsiveNavbar/ResponsiveNavbar.tsx (hunk 1 above)
Comment (ErnestDaDev) [reply src/components/ResponsiveNavbar/ResponsiveNavbar.tsx:102]: Yes, that is true, I noticed it. Thanks for the correction.

**Comment by ErnestDaDev** [comment]
> @ErnestDaDev Thanks for the PR

Sure, You are welcome, Sir. It seems the change is only reflected in src folder but not the dist folder, I think you should recompile it to a new version so it reflects.

**Comment by Abusayid693** [comment]
Yes sorry for the delay it should be now available in v-1.0.16

**Pull request merged**